src/Agent.Worker/Variables.cs

```diff
@@ -34,6 +39,59 @@ public Variables(IHostContext hostContext, IDictionary<string, string> copy)
         public string System_HostType { get { return Get(Constants.Variables.System.HostType); } }
         public string System_TFCollectionUrl { get { return Get(WellKnownDistributedTaskVariables.TFCollectionUrl);  } }
 
+        public void ExpandValues(IDictionary<string, string> target)
```

Comment:
This is the reusable method that is used to expand inputs and endpoint data. It does not perform recursive replacement.